docs/edgeql/sdl/modules.rst

```diff
@@ -0,0 +1,111 @@
+.. _ref_eql_sdl_modules:
+
+=======
+Modules
+=======
+
+This section describes the SDL commands pertaining to
+:ref:`modules <ref_datamodel_modules>`.
+
+
+Example
+-------
+
+Declare an empty module:
+
+.. code-block:: sdl
+
+    module my_module {}
+
+
+Declare a module with some content:
+
+.. code-block:: sdl
+
+    module my_module {
+        type User {
+            required property name -> str;
+        }
+    }
+
+Declare a module implicitly:
+
+.. code-block:: sdl
+
+    type my_module::User {
+        required property name -> str;
+    }
+
+
+Syntax
+------
+
+Define a module corresponding to the :ref:`more explicit DDL
+commands <ref_eql_ddl_modules>`.
+
+.. sdl:synopsis::
+
+    module <ModuleName> "{"
+      [ <schema-declarations> ]
+      ...
+    "}"
+
+Implicit module declaration happens when another SDL declaration uses
+:ref:`fully-qualified name <ref_eql_fundamentals_name_resolution>` for
+the item that is being declared.  The *module* part of the
+*fully-qualified* name implies that a module by that name must also
```

Comment:
What does this mean? It means that before `type xxx::Yyy {}` I must have an empty block `module xxx {}` ?

Reply:
I'll move it closer to the example using this style (which appears later on in the description).

Reply:
I've also reworded it to say that the module will be automatically created based on the fully-qualified name.